server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -2032,11 +2033,14 @@ private void refreshLastCommittedSegmentInfos() {
     }
 
     @Override
-    public void rollTranslogGeneration() throws EngineException {
+    public void rollTranslogGeneration() throws EngineException, TranslogUploadFailedException {
```

Comment:
Can we call `translogManager.rollTranslogGeneration()` from this method instead of having 2 diff implementations?

Reply:
Dans server/src/main/java/org/opensearch/index/engine/InternalEngine.java :

 > @@ -2032,11 +2033,14 @@ private void rafraîchirLastCommitteSegmentInfos() {
     }
 
     @Passer outre
- public void rollTranslogGeneration() lance EngineException {
+ public void rollTranslogGeneration() lance EngineException, TranslogUploadFailedException {
Pouvons-nous appeler translogManager.rollTranslogGeneration() à partir de cette méthode au lieu d'avoir 2 implémentations différentes ?

—
Répondez directement à cet e-mail, consultez-le sur GitHub ou désabonnez-vous .
Vous recevez ceci parce que vous êtes abonné à ce fil de discussion.

Reply:
@sachinpkale There is a slight difference in the exception thrown ultimately from the methods. Checking if there will be any unintentional side effects on making the change you suggested.

Reply:
The rollTranslogGeneration method of the InternalEngine and the InternalTranslogManager classes are identical except the last exception thrown. 
https://github.com/opensearch-project/OpenSearch/blob/e93425dcac1380358b2d493230566c32c636bf9c/server/src/main/java/org/opensearch/index/engine/InternalEngine.java#L2035-L2051
https://github.com/opensearch-project/OpenSearch/blob/e93425dcac1380358b2d493230566c32c636bf9c/server/src/main/java/org/opensearch/index/translog/InternalTranslogManager.java#L92-L111

Prior to the abstraction changes done in #3638 and #3671, the `rollTranslogGeneration()` method in InternalEngine class used to throw EngineException. After the mentioned PRs, in 2.x branch, the differences have come in where the InternalEngine throws EngineException ([code references](https://github.com/opensearch-project/OpenSearch/pull/3671/files#diff-9d23f98d69ed60603d7dd374871a9e86ad5b176ca105f842f4adad7a7c153a50L2025-L2042)) and the InternalTranslogManager throws the TranslogException ([code reference](https://github.com/opensearch-project/OpenSearch/pull/3638/files#diff-015b0808dd36eed04b8395e9be4a91dc2e59459d326c1c57b61cc146406422f6R78-R94)).

I have also validated that it does not matter what exception gets thrown out of the rollTranslogGeneration method as there does not appear to explicit handling of the exception post it is thrown.

Reply:
@FawziEid - Apologies but I do not understand what you have posted. Kindly share your comments in English.